app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -520,6 +529,81 @@ private void showInfoDialog() {
         infoDialog.create().show();
     }
 
+    private void showBalanceUpdateDialog() {
+        Context dialogContext = this;
+
+        AlertDialog.Builder updateDialog = new AlertDialog.Builder(this);
+        LinearLayout layout = new LinearLayout(this);
+        layout.setOrientation(LinearLayout.VERTICAL);
+
+
+        TextView updateTitleView = new TextView(this);
+        updateTitleView.setPadding(20, 20, 20, 20);
+        updateTitleView.setTextSize(settings.getFontSizeMax(settings.getMediumFont()));
+        updateTitleView.setText(R.string.updateBalanceTitle);
+        updateDialog.setCustomTitle(updateTitleView);
+        updateDialog.setTitle(R.string.updateBalanceTitle);
+
+        LinearLayout textLayout = new LinearLayout(this);
+        textLayout.setOrientation(LinearLayout.HORIZONTAL);
+
+        TextView currentTextview = new TextView(this);
+        currentTextview.setPadding(20, 0, 30, 0);
+        currentTextview.setTextSize(settings.getFontSizeMax(settings.getSmallFont()));
+        currentTextview.setTextColor(Color.GRAY);
+        currentTextview.setAutoLinkMask(Linkify.ALL);
+        currentTextview.setText(getString(R.string.currentBalanceSentence, Utils.formatBalance(this, currentBalance, loyaltyCard.balanceType)));
+        textLayout.addView(currentTextview);
+
+        TextView updateTextview = new TextView(this);
+        updateTextview.setPadding(30, 0, 20, 0);
+        updateTextview.setTextSize(settings.getFontSizeMax(settings.getSmallFont()));
+        updateTextview.setTextColor(Color.GRAY);
+        updateTextview.setAutoLinkMask(Linkify.ALL);
+        updateTextview.setText(getString(R.string.newBalanceSentence, Utils.formatBalance(this, currentBalance, loyaltyCard.balanceType)));
+        textLayout.addView(updateTextview);
+
+        layout.addView(textLayout);
+
+        final EditText input = new EditText(this);
+        input.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_DECIMAL);
+        input.setHint(R.string.updateBalanceHint);
+        input.addTextChangedListener(new SimpleTextWatcher() {
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                updateTextview.setText(getString(R.string.newBalanceSentence, Utils.formatBalance(dialogContext, calculateNewBalance(s), loyaltyCard.balanceType)));
+            }
+        });
+        layout.addView(input);
+
+        updateDialog.setView(layout);
+        updateDialog.setPositiveButton(R.string.ok, (dialogInterface, i) -> {
+            dialogInterface.dismiss();
```

Comment:
Closing the dialog closes and then makes re-appear the keyboard. The keyboard probably needs to be hidden before the dismiss.